src/util/is-cloud-or-server.ts

```diff
@@ -0,0 +1,4 @@
+import { GITHUB_CLOUD_API_BASEURL } from "utils/get-github-client-config";
+
+export const isCloudOrServerSubscription = (gitHubAppId: number | undefined) => gitHubAppId ? "server" : "cloud";
+export const isCloudOrServerHost = (host: string) => host === GITHUB_CLOUD_API_BASEURL ? "cloud" : "server";
```

Comment:
/nit
Usually method `isXXX` return `boolean`. We could use `getXXXFromHost` like ( `getGitHubTagFromHost(host)` or `gitHubTagFromHost()` )

Reply:
/nit
And the var name `gitHubVersion`, should it be like `gitHubType`? Isn't `xxxVersion` usually refer to `x.x.x`?

Reply:
I'll update the method name. Regarding version, however, yes it _can_ mean that if we look at it through a narrow 'developer lense' but the literal definition of the word is `a particular form of something differing in certain respects from an earlier form`. Happy to discuss further (especially seeing I can guarantee others on the team would most definitely want to have input on this... @thombergs * cough *) but would say we do it separately to this pr (especially seeing that wording has already been used elsewhere).